### PR TITLE
revisions after code review - policy grouping

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java [hunk 1]
```diff
@@ -214,10 +214,12 @@ public boolean isAppAllowAnonymous(MessageContext messageContext) {
      * @return result : boolean result relevant to registry value
      */
     public boolean isUrlAllowAnonymous(MessageContext messageContext) {
```
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java:216]: this method has been changed entirely in the latest code

File: components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java [hunk 2]
```diff
@@ -214,10 +214,12 @@ public boolean isAppAllowAnonymous(MessageContext messageContext) {
...
-        boolean result;
+        boolean result = false;
```
Comment (nuwand) [inline components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java:217]: Not needed to set to 'false' explicitly. boolean is false by default in java.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/workflow/WorkflowExecutor.java [hunk 3]
```diff
@@ -23,12 +23,13 @@
...
 import org.wso2.carbon.registry.core.utils.UUIDGenerator;
 
+import java.io.Serializable;
 import java.util.List;
 
 /**
  * This is the class that should be extended by each workflow implementation.
  */
-public abstract class WorkflowExecutor {
+public abstract class WorkflowExecutor implements Serializable{
```
Comment (nuwand) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/workflow/WorkflowExecutor.java:32]: Does this need to be Serialized? I don't think this class is ever sent over the network.

File: components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java (hunk 2 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java:217]: this code is changed in the latest code and these line are been removed..

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/workflow/WorkflowExecutor.java (hunk 3 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/workflow/WorkflowExecutor.java:32]: WorkflowExecutor is being used in DTO classes and which are also need to be Serialized. 
Design should be improved (which is a big change) in order to remove the Serializable from this class.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 4]
```diff
@@ -6982,7 +6981,7 @@ public static void savePolicyPartialMappings(Integer policyGroupId,
 	 * @throws AppManagementException if any an error found while saving data to DB
 	 */
 	public static void deletePolicyPartialMappings(Integer policyGroupId, Connection conn)
-			throws AppManagementException {
+			throws SQLException {
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6984]: It is a bad practice to throw an SQL exception from the DAO layer.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 5]
```diff
@@ -7002,30 +7007,34 @@ public static void deletePolicyPartialMappings(Integer policyGroupId, Connection
...
-	 * Get policy group related polices for the given application uuid
+	 * Get policy group related polices for the given application UUID
 	 *
-	 * @param uuid : application uuid
+	 * @param AppUUID
+	 *            : application UUID
 	 * @return : array of policy details objects
 	 */
-	public static JSONArray getPolicyGroupXACMLPoliciesByApplication(String uuid) {
+	public static JSONArray getPolicyGroupXACMLPoliciesByApplication(String AppUUID)
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7016]: Variable names should start from a simple letter.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 6]
```diff
@@ -7002,30 +7007,34 @@ public static void deletePolicyPartialMappings(Integer policyGroupId, Connection
...
-		//query to get policies under each policy group mapped with the application
-		String query = "SELECT DISTINCT MAP.POLICY_GRP_ID AS POLICY_GRP_ID, POL.NAME AS POLICY_GRP_NAME, " +
-				"POL.THROTTLING_TIER AS THROTTLING_TIER, POL.USER_ROLES AS USER_ROLES, " +
-				"COALESCE(POL.URL_ALLOW_ANONYMOUS,'FALSE') AS URL_ALLOW_ANONYMOUS " +
-				"FROM APM_POLICY_GROUP_MAPPING MAP " +
-				"LEFT JOIN APM_POLICY_GROUP POL ON MAP.POLICY_GRP_ID =POL.POLICY_GRP_ID " +
-				"WHERE MAP.APP_ID = (SELECT APP_ID FROM APM_APP WHERE UUID = ?) ";
+		// query to get policies under each policy group mapped with the
+		// application
+		String query =
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7026]: Unnecessary line break.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 7]
```diff
@@ -7034,46 +7043,45 @@ public static JSONArray getPolicyGroupXACMLPoliciesByApplication(String uuid) {
...
+	 * @param conn
+	 *            : sql connection
 	 * @return comma separated list of policy partial names
-	 * @throws AppManagementException on error
+	 * @throws AppManagementException
+	 *             on error
 	 */
-	public static String getPolicyGroupWisePolicyPartials(Integer policyGroupId, Connection conn) throws AppManagementException {
+	public static String getPolicyGroupWisePolicyPartials(Integer policyGroupId, Connection conn)
+			throws SQLException {
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7072]: Throwing an SQL exception form the DAO layer is not a good practice.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 8]
```diff
@@ -7081,20 +7089,28 @@ public static String getPolicyGroupWisePolicyPartials(Integer policyGroupId, Con
 			ps.setInt(1, policyGroupId);
 			rs = ps.executeQuery();
 			while (rs.next()) {
-				//creates the policy partial name list with effect
-				policyNamelist += "[ Name: " + rs.getString("POLICY_PARTIAL_NAME") + ", Effect: " + rs.getString("EFFECT") + "], ";
+				// creates the policy partial name list with effect
+				policyNamelist +=
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7093]: Unnecessary line break.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 5 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7016]: fixed with commit# https://github.com/lahiru823/carbon-appmgt/commit/8b1a171903872cb9922ec5e6085882278b17fda1

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 6 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7026]: default behaviour of formatter

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 8 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7093]: already fixed with a sub sequent commit

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 4 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6984]: This is a private method. (Access modifier has been changed later as 'private' with a subsequent commit)

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 7 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7072]: This is a private method. (Access modifier has been changed later as 'private' with a subsequent commit)

**Pull request merged**